src/dmd/typesem.d

```diff
@@ -3003,6 +3003,16 @@ void resolve(Type mt, const ref Loc loc, Scope* sc, out Expression pe, out Type
             error(loc, "variable `__ctfe` cannot be read at compile time");
             return returnError();
         }
+        if (mt.ident == Id.builtin_va_list) // gcc has __builtin_va_xxxx for stdarg.h
+        {
+            /* Since we don't support __builtin_va_start, -arg, -end, we don't
+             * have to actually care what -list is. A void* will do.
+             * If we ever do care, import core.stdc.stdarg and pull
+             * the definition out of that, similarly to how std.math is handled for PowExp
+             */
+            pt = Type.tvoidptr;
```

Comment:
Because va_list _is_ an internally defined type in gcc, the correct way for gdc would be to use `target.va_listType()` to get the real type.

Reply:
I tried `target.va_listType()`. It fails because `__va_list_tag` is undefined. I'll be happy to fix this to use `va_listType()` if you'll get it to work! In the meantime, `void*` will enable dmd to work.

Reply:
We already pass the `Scope` to `va_listType`, I guess we can get away with something like:
```
if (sc.flags & SCOPE.Cfile)
{
    // Build TypeStruct with fields.
}
else
{
    tvalist = new TypeIdentifier(Identifier.idPool("__va_list_tag")).pointerTo();
}
```

Reply:
> In the meantime, void* will enable dmd to work.

Unless you actually use it?  On 64-bit, you'll get a runtime fault because `void*` does not allocate the actual size of `va_list` that `va_start` and others actually expect.